### PR TITLE
add DeepSeekV3 671b shapes and fix scale shapes in float8_roofline.py

### PR DESCRIPTION
**Pull request by danielvegamyhre**
## Summary
- Add shapes from all Linear layers in DSV3 except output projection and router.gate ([details](https://www.internalfb.com/intern/everpaste/?phabricator_paste_number=2127212819&handle=GOE6xCTIJbRAAA8DAEO7htX2mCxUbsIXAAAz))
    - default M to 81920 (seq_len of 8192 with local_bs of 10) 
- Fix scale shapes which should have scaled dim rounded to nearest multiple of 4, and non scaled dim rounded to nearest multiple of 128

## Results

before (with torch.zeros data and torch.ones scales): 
```
    fwd_M  fwd_K  fwd_N  r_bf16_gemm_s  r_fp8_gemm_s  r_fp8_ovhd_s  ...  b_fp8_gemm_s  b_bf16_e2e_s  b_fp8_e2e_s  b_fp8_e2e_spdp  rb_bf16_gemm_ratio  rb_fp8_gemm_ratio
0   81920   7168   1536       3.08e-03      1.54e-03      5.90e-04  ...      1.45e-03      4.05e-03     3.02e-03            1.34                0.74               1.06
1   81920   1536  24576       1.06e-02      5.29e-03      1.77e-03  ...      5.89e-03      1.53e-02     1.05e-02            1.45                0.75               0.90
2   81920  16384   7168       3.29e-02      1.64e-02      1.67e-03  ...      2.42e-02      5.22e-02     3.68e-02            1.42                0.76               0.68
3   81920   7168    576       1.16e-03      5.78e-04      5.21e-04  ...      1.33e-03      1.85e-03     1.82e-03            1.01                0.48               0.43
4   81920    512  32768       4.70e-03      2.35e-03      2.24e-03  ...      2.63e-03      7.11e-03     7.66e-03            0.93                0.66               0.89
5   81920   7168  18432       3.70e-02      1.85e-02      1.82e-03  ...      2.60e-02      5.77e-02     3.83e-02            1.51                0.73               0.71
6   81920  18432   7168       3.70e-02      1.85e-02      1.82e-03  ...      2.72e-02      5.79e-02     3.90e-02            1.48                0.74               0.68
7   81920   7168  18432       3.70e-02      1.85e-02      1.82e-03  ...      2.63e-02      5.75e-02     3.88e-02            1.48                0.73               0.70
8   81920   7168   2048       4.11e-03      2.06e-03      6.27e-04  ...      2.71e-03      5.34e-03     3.75e-03            1.42                0.67               0.76
9   81920   2048   7168       4.11e-03      2.06e-03      6.27e-04  ...      2.32e-03      5.31e-03     3.82e-03            1.39                0.78               0.88
10  81920   7168   2048       4.11e-03      2.06e-03      6.27e-04  ...      1.96e-03      5.29e-03     3.73e-03            1.42                0.76               1.05

```

after (random data within target dtype range):

```
    fwd_M  fwd_K  fwd_N  r_bf16_gemm_s  ...  b_fp8_e2e_s  b_fp8_e2e_spdp  rb_bf16_gemm_ratio  rb_fp8_gemm_ratio
0   81920   7168   1536       3.08e-03  ...     3.50e-03            0.95                0.91               0.88
1   81920   1536  24576       1.06e-02  ...     1.12e-02            1.03                0.97               0.91
2   81920  16384   7168       3.29e-02  ...     2.60e-02            1.48                0.88               0.88
3   81920   7168    576       1.16e-03  ...     2.46e-03            0.58                0.65               0.55
4   81920    512  32768       4.70e-03  ...     9.74e-03            0.57                0.84               0.73
5   81920   7168  18432       3.70e-02  ...     2.86e-02            1.51                0.89               0.85
6   81920  18432   7168       3.70e-02  ...     3.16e-02            1.35                0.90               0.86
7   81920   7168  18432       3.70e-02  ...     2.91e-02            1.47                0.89               0.84
8   81920   7168   2048       4.11e-03  ...     4.18e-03            1.05                0.88               0.86
9   81920   2048   7168       4.11e-03  ...     4.20e-03            1.05                0.93               0.88
10  81920   7168   2048       4.11e-03  ...     4.18e-03            1.05                0.93               0.88
```

### CODE REVIEW

File: benchmarks/float8/float8_roofline.py [hunk 1]
```diff
@@ -177,8 +177,21 @@ def get_gemm_times(
             scale_a = torch.ones(M, 1, device=device)
             scale_b = torch.ones(1, N, device=device)
         elif mx_recipe_name in ("mxfp8_cublas", "mxfp8_cublas_rceil"):
-            scale_a = torch.ones(M, K // 32, device=device, dtype=torch.float8_e8m0fnu)
-            scale_b = torch.ones(N, K // 32, device=device, dtype=torch.float8_e8m0fnu)
+            M_rounded = round_up(M, 128)
+            K_rounded = round_up(K // 32, 4)
+            N_rounded = round_up(N, 128)
+            scale_a = torch.ones(
```
Comment (vkuzo) [inline benchmarks/float8/float8_roofline.py]: we should init with random numbers instead of ones/zeros, somehow we didnt catch until now, this will make perf more realistic

Comment (vkuzo) [reply benchmarks/float8/float8_roofline.py]: also on l171 and l172

Comment (danielvegamyhre) [reply benchmarks/float8/float8_roofline.py]: good catch, updated

File: benchmarks/float8/float8_roofline.py
Comment (vkuzo) [inline benchmarks/float8/float8_roofline.py]: can we update these ones too? these are the ones that matter the most for getting realistic performance readings
